psse/psse-model/src/main/java/com/powsybl/psse/model/data/RawDataFactory.java

```diff
@@ -0,0 +1,41 @@
+/**
+ * Copyright (c) 2020, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.powsybl.psse.model.data;
+
+import com.powsybl.psse.model.PsseConstants.PsseVersion;
+
+/**
+ * @author Luma Zamarreño <zamarrenolm at aia.es>
+ * @author José Antonio Marqués <marquesja at aia.es>
+ */
+public final class RawDataFactory {
+
+    private RawDataFactory() {
+    }
+
+    public static RawData create(String extension) {
+        // When only extension is given,
+        // We create Raw Data objects that are able to read the case identification
+        // To determine if the file is valid and get its version
+        if (extension.equalsIgnoreCase("rawx")) {
+            return new RawXDataCommon();
+        } else {
+            return new RawDataCommon();
+        }
+    }
+
+    public static RawData create(String extension, PsseVersion version) {
+        if (extension.equalsIgnoreCase("rawx")) {
+            return new RawXData35();
+        } else {
+            if (version == PsseVersion.VERSION_35) {
+                return new RawData35();
+            }
+            return new RawData33();
+        }
+    }
```

Comment:
As explain previously, I would prefer to throw an exception instead of having a fall back. This would prepare the code for another version.

Reply:
Fixed